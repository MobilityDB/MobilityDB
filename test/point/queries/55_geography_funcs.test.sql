-------------------------------------------------------------------------------
--
-- This MobilityDB code is provided under The PostgreSQL License.
--
-- Copyright (c) 2016-2021, Université libre de Bruxelles and MobilityDB
-- contributors
--
-- MobilityDB includes portions of PostGIS version 3 source code released
-- under the GNU General Public License (GPLv2 or later).
-- Copyright (c) 2001-2021, PostGIS contributors
--
-- Permission to use, copy, modify, and distribute this software and its
-- documentation for any purpose, without fee, and without a written 
-- agreement is hereby granted, provided that the above copyright notice and
-- this paragraph and the following two paragraphs appear in all copies.
--
-- IN NO EVENT SHALL UNIVERSITE LIBRE DE BRUXELLES BE LIABLE TO ANY PARTY FOR
-- DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING
-- LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION,
-- EVEN IF UNIVERSITE LIBRE DE BRUXELLES HAS BEEN ADVISED OF THE POSSIBILITY 
-- OF SUCH DAMAGE.
--
-- UNIVERSITE LIBRE DE BRUXELLES SPECIFICALLY DISCLAIMS ANY WARRANTIES, 
-- INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
-- AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS ON
-- AN "AS IS" BASIS, AND UNIVERSITE LIBRE DE BRUXELLES HAS NO OBLIGATIONS TO 
-- PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS. 
--
-------------------------------------------------------------------------------

-------------------------------------------------------------------------------

SELECT ST_AsText(setPrecision(st_LineInterpolatePoint(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.0), 6));
SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.0, true), 6));
SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 1.0, false), 6));
SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.1, true), 6));
-- EMPTY
SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring empty', 0.1, true), 6));
/* Errors */
SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Point(4.35 50.85)', 0.5, true), 6));
SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 2, true), 6));

-------------------------------------------------------------------------------
