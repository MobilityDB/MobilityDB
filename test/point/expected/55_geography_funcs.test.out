SELECT ST_AsText(setPrecision(st_LineInterpolatePoint(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.0), 6));
     st_astext     
-------------------
 POINT(4.35 50.85)
(1 row)

SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.0, true), 6));
     st_astext     
-------------------
 POINT(4.35 50.85)
(1 row)

SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 1.0, false), 6));
         st_astext          
----------------------------
 POINT(37.617222 55.755833)
(1 row)

SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.1, true), 6));
                                                                                                     st_astext                                                                                                     
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 MULTIPOINT(7.222099 51.720243,10.232588 52.525434,13.373925 53.256689,16.634776 53.905679,19.999858 54.464967,23.45005 54.928333,26.962817 55.291077,30.512938 55.550263,34.073528 55.704869,37.617222 55.755833)
(1 row)

SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring empty', 0.1, true), 6));
  st_astext  
-------------
 POINT EMPTY
(1 row)

/* Errors */
SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Point(4.35 50.85)', 0.5, true), 6));
ERROR:  line_interpolate_point: 1st arg isn't a line
SELECT ST_AsText(setPrecision(st_LineInterpolatePoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 2, true), 6));
ERROR:  line_interpolate_point: 2nd arg isn't within [0,1]
